test/test_callbacks.jl

```diff
@@ -44,13 +45,30 @@ simulation = Simulation(
     model,
     SSPRK33(),
     dt = 0.01,
-    tspan = (0.0, 0.02),
+    tspan = (0.0, 0.03),
     callbacks = cb_set,
 )
 run!(simulation)
 
-@test isfile(joinpath(cb_1.filedir, cb_1.filename * ".jld2")) == true
-@test isfile(joinpath(cb_2.filedir, cb_2.filename * ".jld2")) == true
+@info "Testing Simulation Restart"
+simulation = Simulation(
+    model,
+    SSPRK33(),
+    dt = 0.01,
+    tspan = (0.0, 0.03),
+    callbacks = cb_set,
+    restart = Restart(
+        restartfile = joinpath(cb_1.filedir, cb_1.filename * "_0.02.jld2"),
+        end_time = 0.05,
+    ),
+)
+set!(simulation, :swm, h = h, u = u, c = c);
+run!(simulation)
+@test simulation.integrator.t == 0.05
+
+
+@info "Deleting Test Output Files"
+@test isfile(joinpath(cb_1.filedir, cb_1.filename * "_0.01" * ".jld2")) == true
+@test isfile(joinpath(cb_2.filedir, cb_2.filename * "_0.02" * ".jld2")) == true
 
-rm(joinpath(cb_1.filedir, cb_1.filename * ".jld2"))
-rm(joinpath(cb_2.filedir, cb_2.filename * ".jld2"))
+rm(cb_1.filedir, recursive = true)
```

Comment:
Should we remove cb_2 too?

Reply:
I should generalise this to `filedir` for the test-set to avoid confusion, but the recursive delete removes all such temporary files created as part of the callback  / output test process.